### PR TITLE
[Caffe] improve caffe converter

### PR DESCRIPTION
**Pull request by mli**
I create this PR to simplify others to continue working on it. But currently it is not ready for merging. 

Things are changed:

1. docs/how_to/caffe.md for detailed instructions for caffe users to use mxnet
2. added convert_caffe_modelzoo.py to download and convert caffe model zoo models
3. added test_converter.py which tests the accuracy and speed all converted model 

Issues need to be fixed

1. [ ] failed to convert bvlc_alexnet. 
2. [ ] failed to convert resnet due to the lack of Scale layer, related issues #3138 #1473
3. [ ] does not support multiple inputs/outputs models
4. [ ] i didn't test if using a mean.binaryproto for mean instead of a list of rgb mean, see resnet-50
5. [ ] doesn't support PRelu #4139
6. [ ] more code documents, such as we should add docstrings to each function https://www.python.org/dev/peps/pep-0257/
7. [ ] improve code quality in convert_symbol.py and convert_model.py. e.g remove `import caffe` because we can assume users will install protobuf first
8. [ ] (optional)  if an operator is not supported, consider to use `sym.CaffeOp` instead of raise an exception  

to reproduce 1 and 2, just change to `models = ['bvlc_alexnet', 'resnet-50']` in test_converter.py

the bottomline is we should get 1-5 fixed.

### CODE REVIEW

File: tools/caffe_converter/Makefile [hunk 1]
```diff
@@ -1,17 +1,7 @@
...
-endif
-
 all: caffe_parse/caffe_pb2.py
 
 clean:
 	rm caffe_parse/caffe_pb2.py*
 
 caffe_parse/caffe_pb2.py:
-	$(PROTOC) --python_out=./ ./caffe_parse/caffe.proto
+	protoc --python_out=./ ./caffe_parse/caffe.proto
```
Comment (piiswrong) [inline tools/caffe_converter/Makefile]: Why this change?

**Comment by piiswrong** [comment]
2. scale layer can be implemented with broadcast_mul

**Comment by piiswrong** [comment]
I think we have PReLU. Just a matter of adding more switch cases in convert_symbol

**Comment by fengshikun** [comment]
if layer[i].type == 'PReLU':
           type_string = 'mx.symbol.LeakyReLU'
           param_string = "act_type='prelu'"
           need_flatten[name] = need_flatten[mapping[layer[i].bottom[0]]]
     
add such code to `convert_symbol.py` solve the prelu problem

**Comment by sbodenstein** [comment]
Regarding 2: we could follow the approach used by a popular Caffe to TensorFlow converter: [Caffe to TensorFlow](https://github.com/ethereon/caffe-tensorflow/blob/0c5bf53817f89c7e4581ad6b738a4bb19eea96b3/kaffe/transformers.py#L209)

@piiswrong: `broadcast_mul` is not enough, as the Caffe `Scale` layer also has an optional learnable bias parameter, which gets used in resnet.

**Comment by piiswrong** [comment]
@sbodenstein we also have broadcast_add

File: tools/caffe_converter/Makefile (hunk 1 above)
Comment (zihaolucky) [inline tools/caffe_converter/Makefile]: Maybe the protobuf compiler has been installed if the user follows the README first.

**Comment by mli** [comment]
will make a new PR

**Comment by lcwyylcwyy** [comment]
Hi @mli I want to implement CRelu structure. But when process the scale layer, I encounter an issue. 
That is after bn layer, the CRelu use a negative layer to it and concat the two layer[bn, -bn]. Then the subsequent layer is scale. So how to implement the scale layer alone with mxnet?
